### ISSUE TITLE
[SECTION 2800]: SHUTDOWN & SANITIZING - Question

### ISSUE THREAD

**Issue by gitbower0** [arkenfox/user.js#1349]
I'm not sure if this is where to go for questions, as this is not an issue. Apologies if I'm in the wrong spot.

mac/linux
FFv96
AFv96

In my browsing-only FF profile, my only user-overrides include the following:

**user_pref("privacy.clearOnShutdown.offlineApps", true); // [DEFAULT: false]
user_pref("privacy.clearOnShutdown.cookies", true);**

**user_pref("privacy.cpd.offlineApps", true); // [DEFAULT: false]
user_pref("privacy.cpd.cookies", true);**

With the rest of the prefs in this section set to true by default via the repo user.js - Is this overkill? 

Clearly, I prefer a clean slate with each new session. 

Hopefully that made sense - and thank you very much for all the work you do.

**Comment by Thorin-Oakenpants** [comment]
Really? No other overrides? Wow! Holy bat farts! Really? That's encouraging. Mine are all mainly cosmetic things like urlbar stuff.

Currently all cookies and site data is cleared on close (except exceptions) because we use the cookie lifetime polict pref
```js
/* 2801: delete cookies and site data on exit
 * [SETTING] Privacy & Security>Cookies and Site Data>Delete cookies and site data when Firefox is closed ***/
user_pref("network.cookie.lifetimePolicy", 2);
```

In theory your overrides are redundant (one assumes you have no exceptions). But that lifetime pref is going the way of the ~mando~~ dodo, i.e it will be deprecated by Firefox and instead users will use the pref(s) you mention - and I think they will most likely turn on `privacy.sanitize.sanitizeOnShutdown` and I think they will clean up the sanitize on close UI. We'll see.

After first leaving the sea, this is seen as another good move for humans as cookies will no longer be session cookies by default - which can cause problems: e.g
- with uBO's `cookie-remover.js` - scriptlet which sets cookie expiration time to past date - https://github.com/gorhill/uBlock/wiki/Resources-Library#cookie-removerjs-
- it causes service workers to fail

First they going to make those two prefs obey exceptions (not sure on the cpd ones which is manual sanitizing) and that got a [patch](https://bugzilla.mozilla.org/show_bug.cgi?id=1681701#c1) just this week, yet to land
- [1681493](https://bugzilla.mozilla.org/show_bug.cgi?id=1681493) [META] Deprecate and remove network.cookie.lifetimePolicy
- [1681495](https://bugzilla.mozilla.org/show_bug.cgi?id=1681495) - "Delete cookies and site data when Nightly is closed" should update sanitization prefs instead of network.cookie.lifetimePolicy
- [1681498](https://bugzilla.mozilla.org/show_bug.cgi?id=1681498) Migrate users on network.cookie.lifetimePolicy=2 to sanitization prefs
- [1681701](https://bugzilla.mozilla.org/show_bug.cgi?id=1681701) Ensure that shutdown sanitization for cookies and site data adheres to cookie exceptions

tl;dr ...  🎉 one day soon you will have NO overrides 🥇 because arkenfox will be putting those prefs back to true (not sure on the cpd ones just yet)

**Comment by gitbower0** [comment]
> Really? No other overrides? Wow! Holy bat farts! Really? That's encouraging. 

Well..I didn't want to steam-roll over my point of query and add too much to read:) I do have some in sections unrelated i.e. warnings, UX/UI and personal sections. Neat and clean.

> Mine are all mainly cosmetic things like urlbar stuff.

yes, here as well.

> Currently all cookies and site data is cleared on close (except exceptions) because we use the cookie lifetime policy pref

Ah, got it.
 
Just reminded me:     
user_pref("privacy.clearOnShutdown.offlineApps - I'm not clear on what this is(offline), or if that's included in "sanitizing"

Before the switch to v96, I had all sanitize settings set to true(everything) - as well as unrelated prefs - and...wait..
didn't allow any cookies period. Allowed for session when needed for site functionality. Again - browsing profile only.
I may have read on here that disallowing all cookies was a no-go due to ETP/TCP.
Much cleaner this way. Easier. 

> tl;dr ... 🎉 one day soon you will have NO overrides 🥇 because arkenfox will be putting those prefs back to true (not sure on the cpd ones just yet)

That's awesome. I'll be here...waiting...but not in a creepy way...but maybe a little. 

Thanks for your time, and the additional info.

**Comment by Thorin-Oakenpants** [comment]
> user_pref("privacy.clearOnShutdown.offlineApps - I'm not clear on what this is(offline), or if that's included in "sanitizing"

https://github.com/arkenfox/user.js/blob/10044fcaf77bfe19f2b90c606232f0a9e83401c9/user.js#L806

Cookies is used for logins, but some login flows or login types also require offlineApp items (probably IDB) which is why that was also flipped - see #1291 

> Before the switch to v96

up until about v95, I blocked all cookies everywhere and this blocked all site data (cookies, SWers, UDB etc) and I never had anything to clean. And I didn't sanitize cookies, instead I kept 4 for logins (by adding an allow exception) and OK'ed 6 for sessions only (by adding exceptions as `Allow for session`). My disk was hardly ever touched. But this is incompatible with dFPI also known as Total Cookie Protection

Just blanket allowing cookies (with Tracking Protection exceptions) and they're all isolated/partitioned to first party reduces breakage and doesn't compromise anything

> I'll be here...waiting...but not in a creepy way...but maybe a little

Excellent!

**Comment by Thorin-Oakenpants** [comment]
> or if that's included in "sanitizing"

yes, it is the "site data" part of 
- [SETTING] Privacy & Security>Cookies and Site Data>Delete cookies and site data when Firefox is closed

**Comment by gitbower0** [comment]
Hello again, gents. 

Quick question, and I'll return to my dwelling for another few months.

My overrides for close/during session for many moons. 
Would like to know if/when I've reached the point of redundancy. For this profile here - a clean slate each time warms the cockles of my heart.

As always - thank you again for MGRA! making Gecko great again lol

Seriously, thank you for the work. 

/*** [SECTION 2800]: SHUTDOWN & SANITIZING ***/
user_pref("_user.js.parrot", "2800 syntax error: the parrot's bleedin from the butt-parts");

user_pref("privacy.clearOnShutdown.cache", true);     // [DEFAULT: true]
user_pref("privacy.clearOnShutdown.downloads", true); // [DEFAULT: true]
user_pref("privacy.clearOnShutdown.formdata", true);  // [DEFAULT: true]
user_pref("privacy.clearOnShutdown.history", true);   // [DEFAULT: true]
user_pref("privacy.clearOnShutdown.sessions", true);  // [DEFAULT: true]
user_pref("privacy.clearOnShutdown.offlineApps", true); // [DEFAULT: false]
user_pref("privacy.clearOnShutdown.cookies", true);
user_pref("privacy.cpd.cache", true);    // [DEFAULT: true]
user_pref("privacy.cpd.formdata", true); // [DEFAULT: true]
user_pref("privacy.cpd.history", true);  // [DEFAULT: true]
user_pref("privacy.cpd.sessions", true); // [DEFAULT: true]
user_pref("privacy.cpd.offlineApps", true); // [DEFAULT: false]
user_pref("privacy.cpd.cookies", true);

**Comment by fxbrit** [comment]
I think you don't need any overrides if what you want is a clean state on each new browser opening: unless you manually set exceptions everything will be cleared on close.

**Comment by Thorin-Oakenpants** [comment]
I don't understand your question.

**Comment by gitbower0** [comment]
> I think you don't need any overrides if what you want is a clean state on each new browser opening: unless you manually set exceptions everything will be cleared on close.

That's what I was looking for. 
Appreciate it. Thanks.

**Comment by gitbower0** [comment]
> I don't understand your question.

That's because today is the day after Saturday night and my brain is mush. That's on me. 

Thank you again.
